### PR TITLE
feat: handle pvc/pv storageclass replacement

### PR DESCRIPTION
**Pull request by pudong-sh**
replace StorageClass name when restoring

### CODE REVIEW

File: examples/storageclass_restore.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+apiVersion: velero.io/v1
+kind: Restore
+metadata:
+  name: ys1000-storageclass-restore
+  namespace: velero
+  annotations:
+    jibudata.com/pvc-storageclass-plugin: |
+      # add 1+ key-value pairs here, where the key is the old StorageClass name and the value is the
+      # new StorageClass name.
+      {"old-storageclass-name": "new-storageclass-name"}
```
Comment (27149chen) [inline examples/storageclass_restore.yaml:10]: usually a map is used to store different kvs, like {"name":"a", "age": 10}, but storageclass mappings here is a list of sc pairs, so I think a list is better, such as: "old-1:new-1,old-2:new-2"

File: internal/restore/pv_storageclass_restore.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	p.log.Info("PV StorageClass Restore plugin")
+
+	var pv corev1.PersistentVolume
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(input.Item.UnstructuredContent(), &pv); err != nil {
+		return nil, errors.Wrap(err, "unable to convert pv from runtime.Unstructured")
+	}
+	p.log.Infof("PV StorageClass Restore, pv: %s", pv.Name)
+
+	restore := input.Restore
+	scString, ok := restore.Annotations[util.PVStorageClassPlugin]
```
Comment (27149chen) [inline internal/restore/pv_storageclass_restore.go:56]: check if restore.Annotations is nil

File: internal/restore/pvc_storageclass_restore.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+	p.log.Info("PVC StorageClass Restore plugin")
+
+	var pvc corev1.PersistentVolumeClaim
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(input.Item.UnstructuredContent(), &pvc); err != nil {
+		return nil, errors.Wrap(err, "unable to convert pvc from runtime.Unstructured")
+	}
+	p.log.Infof("PVC StorageClass Restore, pvc: %s/%s", pvc.Namespace, pvc.Name)
+
+	restore := input.Restore
+	scString, ok := restore.Annotations[util.PVCStorageClassPlugin]
```
Comment (27149chen) [inline internal/restore/pvc_storageclass_restore.go:56]: check if restore.Annotations is nil

**Review by jerry-jibu — APPROVED** [review]
lgtm, thanks.

**Pull request merged**